tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,740 @@
+        if (summaryMode) {
+            printDescriptorBlocks(currentCount);
+        }
+        count.getAndIncrement();
+    }
+
+    /**
+     * Copies a Descriptor into a temporary one for it can be cloned into the list of descriptors
+     *
+     * @param original The original descriptor that is continually modified
+     */
+    private void copyAddResetDescriptor(Descriptor original) {
+        Descriptor temp = new Descriptor(original);
+        descriptors.add(temp);
+        original.reset();
+    }
+
+    /**
+     * Determines if the current descriptor block is missing any required values
+     * and then if it is, prints out a warning message and sets that required value
+     * and the optional values to null
+     *
+     * @param count The index of the current descriptor block
+     */
+    private void findMissingRequiredValuesAndSetValues(int count, Descriptor descriptor) {
+        boolean missingRequiredValue = false;
+        if (descriptor.getInputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlock(count, "missing input realm location"));
+            missingRequiredValue = true;
+        }
+        if (descriptor.getOutputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlock(count, "missing output realm location"));
+            missingRequiredValue = true;
+        }
+        if (descriptor.getFileSystemRealmName() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlock(count, "missing new filesystem realm name"));
+            missingRequiredValue = true;
+        }
+        if(descriptor.getHashCharset() == null) {
+            descriptor.setHashCharset(StandardCharsets.UTF_8);
+        }
+        if(descriptor.getHashEncoding() == null) {
+            descriptor.setHashEncoding(Encoding.BASE64);
+        }
+        if(descriptor.getEncoded() == null) {
+            descriptor.setEncoded(true);
+        }
+        if(descriptor.getLevels() == 0) {
+            descriptor.setLevels(2);
+        }
+        if (missingRequiredValue) {
+            descriptor.reset();
+        }
+    }
+
+    /**
+     * Handles creating the Elytron filesystem-realm from the descriptor array
+     *
+     * @throws Exception Exception to be handled by Elytron Tool
+     */
+    private void createFileSystemRealm() throws Exception {
+        Security.addProvider(new WildFlyElytronProvider());
+        for (Descriptor descriptor : descriptors) {
+            System.out.println("Creating encrypted realm for: " + descriptor.getInputRealmLocation());
+            if (descriptor.getFileSystemRealmName() == null || descriptor.getInputRealmLocation() == null || descriptor.getOutputRealmLocation() == null) {
+                continue;
+            }
+            CredentialStore credentialStore;
+            if ((descriptor.getCredentialStore() == null) == (descriptor.getSecretKeyAlias() == null)) {
```

Comment:
&& instead of ==

Reply:
I did the `==` here to check if either both are not `null` or both are `null`.
If only 1 is defined but not the other, and error is thrown: https://github.com/wildfly-security/wildfly-elytron/pull/1676/files/b1443435ea996fc76917a3bd58c7f14852c30819#diff-ebe53241afd70e1e5743d6db5b39150461c335abce6c083cafbf6aae4b4c1e2dR650